### ISSUE TITLE
Attempt to edit "data" results in editing forbidden.html page

### ISSUE THREAD

**Issue by dblume** [dokuwiki/dokuwiki#3956]
**Edit**: Issue is _not_ a regression from Hogfather. I wasn't logged in to Hogfather wiki when I tested for regression. I'm leaving the report alone, though, so the thread still makes sense.

As an admin, when my wiki was "Hogfather", if I navigated to "/bin" or "/data" the wiki would say

> This topic does not exist yet
> You've followed a link to a topic that doesn't exist yet. If permissions allow, you may create it by clicking on **Create this page**.

However, the icon where "Create this page" would be is only "Show page source", so it was OK, I couldn't screw things up.

I've upgraded to "Jack Jackrum", and now if I navigate to "/bin" or "/data", I get the same error message, but the "Create this page" button _works_, and if I press it, the page I'm editing is "forbidden.html?do=edit", and things begin to become problematic. (Yes, I actually wanted to make a page called "data".)

In my opinion, this new behavior is a regression in "Jack Jackrum" from "Hogfather". (I skipped "Igor" so don't know if the regression is in Igor.)

**Comment by splitbrain** [comment]
What's your webserver? And how does your webserver setup to deny access to these directories look like?

I don't think anything changed in that regard in Igor nor Jackrum...

**Comment by splitbrain** [comment]
> However, the icon where "Create this page" would be is only "Show page source", so it was OK

That behavior was already not okay. Denying access to these directories is not part of DokuWiki (beyond providing .htaccess files) but needs to be done by the web server. DokuWiki core doesn't mind creating a page named `data` as it would be stored as `data/pages/data.txt` which is perfectly fine.

A correct setup would look something like this: https://www.dokuwiki.org/data

**Comment by fiwswe** [comment]
@splitbrain: Now that you mention it: I can access [https://www.dokuwiki.org/bin](https://www.dokuwiki.org/bin)! It shows a directory listing and I can access the `.php` files though it seems the PHP code is not executed.

```
$ curl --head --url 'https://www.dokuwiki.org/bin/'
HTTP/1.1 200 OK
Date: Sat, 15 Apr 2023 10:16:46 GMT
Server: Apache
Content-Type: text/html;charset=UTF-8

$ 
```

Same with: https://www.dokuwiki.org/vendor/

**Comment by splitbrain** [comment]
@fiwswe ouch! Thanks for pointing that out. The .htaccess files were missing there because of a fluke in how I deploy dokuwiki.org. Fixed.

**Comment by dblume** [comment]
@splitbrain, my environment: Shared hosting at dreamhost,
```
$ /usr/sbin/apache2 -v
Server version: Apache/2.4.29 (Ubuntu)
Server built:   2022-06-23T12:51:37
```
**CORRECTION**: It was _not_ a regression from Hogfather. When I'd tested for regression, I wasn't logged in at the Hogfather wiki, and so had read-only access, as expected. My bad.

So they both allow me to proceed and "edit" forbidden.html?do=edit

So, maybe I have .htaccess deny access bin/ and data/. I'll see what your current documentation suggests.

But also: this was confusing, and I can still reproduce it:

1. Enter in URL bar "wiki.dlma.com/data", 
2. See text saying page doesn't exist, but **URL remains "wiki.dlma.com/data"** (but don't notice the title has changed to forbidden.html")
3. Click "Create this page" (if logged in)

**Observed**: New page is "forbidden.html?do=edit" which seemingly will create forbidden.html files.

**Expected**: Either new page is "data?do=edit" or it's "forbidden.html" but not editable as a wiki page.

Thing is, I thought I was making a new "data" page. It took a while for me to notice the redirect to forbidden.html, and that it was being treated as an editable wiki page.

Indeed, my history shows these two entries, so the "forbidden" pages were written last night:
```
  445  rm meta/forbidden.html.*
  446  rm attic/forbidden.html.1681543*
```

I'm leaving this open for now, because I think the URL redirection to forbidden.html at the time of creation and being able to create/edit forbidden.html is confusing.

If the true fix is to deny via .htaccess or something like that, I will accept that resolution.

Aside: My reporting is imperfect because I'm untangling this experience with another crash in plugins/semantic, which doesn't gracefully handle deleted pages (I deleted the new forbidden.html page by saving an empty version of it). That one's obviously not your fault, of course, and I fixed locally, but you know how it is when you're shaving yaks and trying to figure out which symptom goes with which cause.

**Comment by fiwswe** [comment]
@dblume 
If you are using Apache httpd and you have the default `.htaccess` files then you should be getting 403 status codes, not some redirection, etc.

If that does not happen, then maybe the Apache config does not allow your `.htaccess` files to be used (see [AllowOverride](https://httpd.apache.org/docs/2.4/en/mod/core.html#allowoverride)).

But I do see this happening from here:
```
$ curl --head --url 'https://wiki.dlma.com/data'   
HTTP/2 403 
date: Sat, 15 Apr 2023 18:53:08 GMT
server: Apache
vary: Cookie,User-Agent
expires: Thu, 19 Nov 1981 08:52:00 GMT
cache-control: no-store, no-cache, must-revalidate
pragma: no-cache
set-cookie: DokuWiki=45d02418afd731e5ee5fec139139f463; path=/; secure; HttpOnly
set-cookie: DW7fa065a06cb74b536c124cfbe56ac6d3=deleted; expires=Thu, 01 Jan 1970 00:00:01 GMT; Max-Age=0; path=/; secure; HttpOnly
content-type: text/html; charset=utf-8

$ 
```

This is consistent with your observation:
> 2\. See text saying page doesn't exist, but **URL remains "wiki.dlma.com/data"** (but don't notice the title has changed to forbidden.html")

However this is strange:
> 3. Click "Create this page" (if logged in)

A normal default 403 error is a very simple page that does not contain any "Create this page" links. You seem to have some setting that overrides your 403 error page to display a wiki page. I don't think there is any DokuWiki setting that does this So check your Apache httpd config for [ErrorDocument](https://httpd.apache.org/docs/2.4/en/mod/core.html#errordocument) statements. They may be the cause for the confusion?

**Comment by dblume** [comment]
Thank you. My .htaccess matches the .htaccess.dist in the download, except I uncommented the RewriteRules for nice URLs. I did a global search of the site, and there's no mention of forbidden.html anywhere. So, I'm thinking this is resulting from my using Shared Webhosting at DreamHost, where they have Apache configs I'm not allowed to see.

I could add new ErrorDocument rules, but as it's already redirecting anyway, it's easy for me to just make a real /forbidden.html file.

Thanks for your time, and so sorry about the bad report. I really appreciate DokuWiki.
